### ISSUE TITLE
make_extended_trapezoid() creates gradient event with slew rate bigger than max_slew_rate without warning

### ISSUE THREAD

**Issue by h3lg3** [imr-framework/pypulseq#216]
**Describe the bug**
Creating an extended trapezoid with arbitrary amplitudes and times may results in a gradient with a slew rate bigger than max_slew_rate without warning. However, if flag convert_to_arbitrary is set true, a violation of max_slew_rate is reported. 

**To Reproduce**
```
import pypulseq as pp
import numpy as np
system = pp.Opts()

# Set max slew to 40 T/m/s
system.max_slew = 40 * system.gamma  # Hz/m/s

gs_rise_time = 500e-6
gs_fall_time = 500e-6
gs_flat_time = 440e-6
gs_amplitude = 851063.829787234

grf_rise_time = 200e-6
grf_fall_time = 200e-6
grf_flat_time = 2.5e-3
grf_amplitude = 256000
gss_times = np.cumsum([0, 
                        gs_rise_time, 
                        gs_flat_time,
                        gs_fall_time - grf_rise_time, 
                        grf_flat_time,
                        gs_rise_time - grf_fall_time,
                        gs_flat_time, 
                        gs_fall_time])

gss_amps = np.array([0, 
                        gs_amplitude, 
                        gs_amplitude, 
                        grf_amplitude, 
                        grf_amplitude, 
                        gs_amplitude, 
                        gs_amplitude, 
                        0])

gss_spoil_add = pp.make_extended_trapezoid(channel = 'z', amplitudes = gss_amps, times = gss_times, system = system)

g = pp.points_to_waveform(times=gss_times, amplitudes=gss_amps, grad_raster_time=system.grad_raster_time)

slew = np.squeeze(np.subtract(g[1:], g[:-1]) / system.grad_raster_time)
print(f"Max slew rate of gradient: {round(max(slew)/system.gamma, 2)} T/m/s")

```

**Expected behavior**
I would expect to get a warning if the gradient exceeds the maximum allowed slew rate. Maybe the same check for slew rate and amplitude can be performed as for make_arbitrary_grad()? Or am I misunderstanding the purpose of make_extended_trapezoid()?

```
g = points_to_waveform(
      times=times, amplitudes=amplitudes, grad_raster_time=system.grad_raster_time
  )

slew = np.squeeze(np.subtract(g[1:], g[:-1]) / system.grad_raster_time)

if max(abs(slew)) >= max_slew:
  raise ValueError(f"Slew rate violation {max(abs(slew)) / max_slew * 100}")
if max(abs(g)) >= max_grad:
  raise ValueError(f"Gradient amplitude violation {max(abs(g)) / max_grad * 100}")
```

**Desktop (please complete the following information):**
 - OS: [e.g. Win11]
- `pypulseq` version: [e.g. 1.4.2]

**Comment by FrankZijlstra** [comment]
As with `make_trapezoid`, I agree that this should be checked as you suggested (at one point I had this check in my local repo, don't know why I never committed it..).

The reason Maxim Zaitsev gave for not checking this in the Matlab Pulseq is that he considers `max_slew` not to be a strict limit (as it isn't on the scanner). You can go slightly over the specified maximum slew rate without issue (usually). Personally, I don't find this particularly appealing. I'd rather have a hard limit that gives an error when exceeded, because the user can always raise the limit slightly if the scanner can handle it.

**Comment by FrankZijlstra** [comment]
Resolved in #217.
